test/eventmesh.Etcd3EventMeshServer.spec.js

```diff
@@ -0,0 +1,351 @@
+'use strict';
+
+const Promise = require('bluebird');
+const CONST = require('../broker/lib/constants');
+const eventmesh = require('../eventmesh');
+const {
+  Etcd3
+} = require('etcd3');
+
+
+describe('eventmesh', () => {
+  describe('Etcd3EventMeshServer', () => {
+    let prefixWatcherStub, keyWatcherStub;
```

Comment:
There is incorrect restoring of stub which is hindering in other test cases. Can you do the following?

```
    let sandbox, valueStub, stringStub, jsonStub, putstub, getstub, prefixWatcherStub, keyWatcherStub;
    before(() => {
      sandbox = sinon.sandbox.create();
      valueStub = sandbox.stub();
      stringStub = sandbox.stub();
      jsonStub = sandbox.stub();
      putstub = sandbox.stub(Etcd3.prototype, 'put', () => {
        return {
          value: (val) => Promise.resolve(valueStub(val))
        };
      });
      getstub = sandbox.stub(Etcd3.prototype, 'get', () => {
        return {
          json: () => Promise.resolve(jsonStub()),
          string: () => Promise.resolve(stringStub()),
        };
      });

      prefixWatcherStub = sandbox.stub().returns({
        create: () => Promise.resolve({
          on: () => Promise.resolve('prefixWatcherStubResponse')
        }),
      });
      keyWatcherStub = sandbox.stub().returns({
        create: () => Promise.resolve({
          on: () => Promise.resolve('keyWatcherStubResponse')
        }),
      });
      sandbox.stub(Etcd3.prototype, 'watch', () => {
        return {
          prefix: prefixWatcherStub,
          key: keyWatcherStub
        };
      });
    });

    afterEach(function () {
      valueStub.reset();
      prefixWatcherStub.reset();
      keyWatcherStub.reset();
      putstub.reset();
      getstub.reset();
      jsonStub.reset();
      stringStub.reset();
    });

    after(function () {
      sandbox.restore();
    });

```

It works for me. Just replace the code and see if it works.

Reply:
Updated